demos/projects/ST/stm32h745i-disco/cm7/port/azure_iot_flash_platform.c

```diff
@@ -0,0 +1,229 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+#include <string.h>
+#include "azure_iot_flash_platform.h"
+#include "azure_iot_flash_platform_port.h"
+#include "stm32h7xx_hal.h"
+/* Logging */
+#include "azure_iot.h"
+#include "azure/core/az_base64.h"
+#include "mbedtls/md.h"
+
+#define azureiotflashH745_WORD_SIZE    32
+#define azureiotflashSHA_256_SIZE      32
+
+static uint8_t ucPartitionReadBuffer[ 32 ];
+static uint8_t ucDecodedManifestHash[ azureiotflashSHA_256_SIZE ];
+static uint8_t ucCalculatedHash[ azureiotflashSHA_256_SIZE ];
+
+static AzureIoTResult_t prvBase64Decode( uint8_t * base64Encoded,
+                                         size_t ulBase64EncodedLength,
+                                         uint8_t * pucOutputBuffer,
+                                         size_t bufferLen,
+                                         size_t * outputSize )
+{
+    az_result xCoreResult;
+
+    az_span encodedSpan = az_span_create( base64Encoded, ulBase64EncodedLength );
+
+    az_span outputSpan = az_span_create( pucOutputBuffer, bufferLen );
+
+    if( az_result_failed( xCoreResult = az_base64_decode( outputSpan, encodedSpan, ( int32_t * ) outputSize ) ) )
+    {
+        AZLogError( ( "az_base64_decode failed: core error=0x%08x", xCoreResult ) );
+        return eAzureIoTErrorFailed;
+    }
+
+    AZLogInfo( ( "Unencoded the base64 encoding\r\n" ) );
+
+    return eAzureIoTSuccess;
+}
+
+AzureIoTResult_t AzureIoTPlatform_Init( AzureADUImage_t * const pxAduImage )
+{
+    pxAduImage->ulCurrentOffset = 0;
+    pxAduImage->ulImageFileSize = 0;
+
+    static FLASH_EraseInitTypeDef xEraseInitStruct;
+    uint32_t ulPageError;
+    FLASH_OBProgramInitTypeDef xOptionBytes;
+    AzureIoTResult_t xResult = eAzureIoTSuccess;
+
+    /* Clear OPTVERR bit set on virgin samples. */
+    __HAL_FLASH_CLEAR_FLAG( FLASH_FLAG_OPERR );
+    /* Get current optionbytes configuration */
+    HAL_FLASHEx_OBGetConfig( &xOptionBytes );
+
+    pxAduImage->xUpdatePartition = ( xOptionBytes.USERConfig & OB_SWAP_BANK_ENABLE ) == OB_SWAP_BANK_DISABLE ?
```

Comment:
If we're calculating this twice, should we save it in a var to reuse? `( xOptionBytes.USERConfig & OB_SWAP_BANK_ENABLE ) == OB_SWAP_BANK_DISABLE`